### PR TITLE
fix(header)!: tendermint compatible json marshall

### PR DESCRIPTION
**Pull request by zvolin**
Fixes https://github.com/celestiaorg/celestia-node/issues/3918

**Please note: this PR is BREAKING** as header encoding has changed.

### CODE REVIEW

File: header/header.go [hunk 1]
```diff
@@ -227,56 +226,19 @@ func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
-// MarshalJSON marshals an ExtendedHeader to JSON. The ValidatorSet is wrapped with amino encoding,
-// to be able to unmarshal the crypto.PubKey type back from JSON.
+// MarshalJSON marshals an ExtendedHeader to JSON.
+// Uses tendermint encoder for tendermint compatiblity.
```
Comment (cristaloleg) [inline header/header.go:230]: Lint failure on `compatibility`

File: header/header.go [hunk 2]
```diff
@@ -227,56 +226,19 @@ func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
...
-	return json.Marshal(&struct {
-		RawHeader    json.RawMessage `json:"header"`
-		ValidatorSet json.RawMessage `json:"validator_set"`
-		*Alias
-	}{
-		ValidatorSet: validatorSet,
-		RawHeader:    rawHeader,
-		Alias:        (*Alias)(eh),
-	})
+	return tmjson.Marshal(*eh)
```
Comment (Wondertan) [inline header/header.go:232]: It shouldn't be necessary to pass the value unless there is some quirk in the `tmjson`. Have you tried without `*`?

Comment (zvolin) [reply header/header.go]: it looks like it was the same case with recursive call as in unmarshal. Just dereferencing hidden that. Switched to alias to be clear what issue is solved there

Comment (Wondertan) [reply header/header.go]: I see

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**